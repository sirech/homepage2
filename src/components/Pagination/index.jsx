import React from 'react'
import PropTypes from 'prop-types'
import Link from 'gatsby-link'

import Container from '../Container'

const url = index => (index === 1 ? '/blog/' : `/blog/${index}`)

const previous = index => (
  <Link className="page-link" to={url(index - 1)}>
    Back
  </Link>
)

const next = index => (
  <Link className="page-link" to={url(index + 1)}>
    Next
  </Link>
)

const Pagination = ({ index, pageCount }) => (
  <Container>
    <nav className="mt-4">
      <ul className="pagination justify-content-end">
        <li className="page-item">{previous(index)}</li>
        <li className="page-item">{next(index)}</li>
      </ul>
    </nav>
  </Container>
)

Pagination.propTypes = {
  index: PropTypes.number,
  pageCount: PropTypes.number,
}

export default Pagination
