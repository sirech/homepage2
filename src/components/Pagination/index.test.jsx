import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Pagination from './index'

describe('components', () => {
  describe('Pagination', () => {
    it('renders correctly', () => {
      const component = shallow(<Pagination index={3} pageCount={5} />)
      expect(toJson(component)).toMatchSnapshot()
    })

    it('renders correctly if it links to the first page', () => {
      const component = shallow(<Pagination index={2} pageCount={5} />)
      expect(toJson(component)).toMatchSnapshot()
    })
  })
})
