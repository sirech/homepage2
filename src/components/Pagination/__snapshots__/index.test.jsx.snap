// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Pagination renders correctly 1`] = `
<Container>
  <nav
    className="mt-4"
  >
    <ul
      className="pagination justify-content-end"
    >
      <li
        className="page-item"
      >
        <Component
          className="page-link"
          to="/blog/2"
        >
          Back
        </Component>
      </li>
      <li
        className="page-item"
      >
        <Component
          className="page-link"
          to="/blog/4"
        >
          Next
        </Component>
      </li>
    </ul>
  </nav>
</Container>
`;

exports[`components Pagination renders correctly if it links to the first page 1`] = `
<Container>
  <nav
    className="mt-4"
  >
    <ul
      className="pagination justify-content-end"
    >
      <li
        className="page-item"
      >
        <Component
          className="page-link"
          to="/blog/"
        >
          Back
        </Component>
      </li>
      <li
        className="page-item"
      >
        <Component
          className="page-link"
          to="/blog/3"
        >
          Next
        </Component>
      </li>
    </ul>
  </nav>
</Container>
`;
